django/contrib/contenttypes/management/__init__.py

```diff
@@ -102,32 +116,15 @@ def update_contenttypes(app_config, verbosity=2, interactive=True, using=DEFAULT
     except LookupError:
         return
 
-    if not router.allow_migrate_model(using, ContentType):
-        return
-
-    ContentType.objects.clear_cache()
     # Always clear the global content types cache.
     if apps is not global_apps:
         global_apps.get_model('contenttypes', 'ContentType').objects.clear_cache()
```

Comment:
As `remove_stale_contenttypes` doesn't deal rely on migration generated apps you should be able to remove the three lines above.